src/Compilers/CSharp/Portable/Binder/NameofBinder.cs

```diff
@@ -19,19 +21,25 @@ namespace Microsoft.CodeAnalysis.CSharp
     /// </summary>
     internal sealed class NameofBinder : Binder
     {
-        private readonly SyntaxNode _nameofArgument;
+        private readonly ExpressionSyntax _nameofArgument;
```

Comment:
This always had to be an ExpressionSyntax.  Making strongly typed and to match the sig of OpenTypeVisitor.VisitThis